hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/googlebigquery/avro/GoogleBigQueryAvroToProtoConverter.java

```diff
@@ -0,0 +1,143 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro;
+
+import com.google.cloud.bigquery.storage.v1.TableSchema;
+import com.google.cloud.bigquery.storage.v1.ToProtoConverter;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.Message;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.StreamSupport;
+import org.apache.avro.generic.GenericRecord;
+
+public class GoogleBigQueryAvroToProtoConverter implements ToProtoConverter<GenericRecord> {
+  @Override
+  public List<DynamicMessage> convertToProtoMessage(
+      Descriptors.Descriptor protoSchema,
+      TableSchema tableSchema,
+      Iterable<GenericRecord> inputObjects,
+      boolean ignoreUnknownFields) {
+    return StreamSupport.stream(inputObjects.spliterator(), false)
+        .map(it -> convertToProtoMessage(protoSchema, it))
+        .toList();
+  }
+
+  public DynamicMessage convertToProtoMessage(
+      Descriptors.Descriptor protoSchema, GenericRecord inputObject) {
+    return createMessage(protoSchema, inputObject);
+  }
+
+  private DynamicMessage createMessage(
+      Descriptors.Descriptor protoSchema, GenericRecord inputObject) {
+    DynamicMessage.Builder messageBuilder = DynamicMessage.newBuilder(protoSchema);
+
+    protoSchema
+        .getFields()
+        .forEach(
+            field -> {
+              convertSimpleField(inputObject, field, messageBuilder);
+            });
+
+    return messageBuilder.build();
+  }
+
+  private void convertSimpleField(
+      GenericRecord inputObject,
+      Descriptors.FieldDescriptor field,
+      Message.Builder messageBuilder) {
+    String fieldName = field.getName();
+    if (!inputObject.hasField(fieldName)) {
+      return;
+    }
+
+    Object fieldValue = inputObject.get(fieldName);
+    if (fieldValue == null) {
+      return;
+    }
+    if (field.isRepeated()) {
+      if (fieldValue instanceof Map<?, ?>) {
+        for (Map.Entry<?, ?> el : ((Map<?, ?>) fieldValue).entrySet()) {
+          DynamicMessage.Builder entryBuilder = DynamicMessage.newBuilder(field.getMessageType());
```

Comment:
could you put it in a method with descriptive name or a comment what it is actually doing?